pkg/activator/handler/requestevent_handler_test.go

```diff
@@ -69,3 +69,35 @@ func TestRequestEventHandler(t *testing.T) {
 		t.Errorf("out.Time.Sub(in.Time) = %v, want > %v", out.Time.Sub(in.Time), timeOffset)
 	}
 }
+
+func BenchmarkRequestEventHandler(b *testing.B) {
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+	reqCtx := util.WithRevID(context.Background(), types.NamespacedName{Namespace: testNamespace, Name: testRevName})
+
+	reqChan := make(chan network.ReqEvent, 100)
+	defer close(reqChan)
+	go func() {
+		// Just consume all the events.
+		for range reqChan {
```

Comment:
this seems like it could end up maybe being slightly unrepresentative, since it will compare reading as fast as possible from the channel whereas in practice we can only read from the channel as fast as our stat calculation takes iyswim. I'm not sure how much it matters (probably zero) since the channel is decently buffered, though 🤷. If you think it's fine then so do I :).

Reply:
I did think the same, though I wanted to keep this local to this handler for now, so we have a clear undistorted picture. If the other handler wouldn't win as clearly, I'd probably add it :joy: